erdkotlin/src/main/java/com/elrond/erdkotlin/domain/dns/ComputeDnsAddressUsecase.kt

```diff
@@ -0,0 +1,70 @@
+package com.elrond.erdkotlin.domain.dns
+
+import com.elrond.erdkotlin.domain.account.models.Account
+import com.elrond.erdkotlin.domain.wallet.models.Address
+import org.bouncycastle.jcajce.provider.digest.Keccak
+import org.bouncycastle.util.encoders.Hex
+import java.nio.charset.StandardCharsets
+
+// Returns the compatible DNS address
+//
+// The implementation is based on
+// - https://github.com/ElrondNetwork/elrond-sdk/blob/52a0ba4414f96068a280b936bd3b456d0b2dd557/erdpy/dns.py
+// - https://github.com/ElrondNetwork/elrond-sdk/blob/52a0ba4414f96068a280b936bd3b456d0b2dd557/erdpy/contracts.py
+internal class ComputeDnsAddressUsecase {
+
+    fun execute(username: String): Address {
+        val hash = nameHash(username)
```

Comment:
maybe add some checks for the username? only lowercase characters and so on :D

Reply:
Can you please provide me the exact list of checks ?
I cannot find it.

Reply:
Username constraints:
- alphanumeric only
- lowercase
- between 3 and 25 characters (without `.elrond` suffix)
- must end with `.elrond` suffix

Reply:
Thanks Dragos :)
The username check is now implemented
I have also push several unit tests for it.